test/test_mail.py

```diff
@@ -209,6 +225,24 @@ def test_summary_mail_from_address(self):
         self.assertTrue(
             "From: rootolv" in mail_txt, "missing From: in %s" % mail_txt)
 
+    @patch("unattended_upgrade.main_inner")
+    def test_exception(self, mock_main_inner):
+        exception_string = "Test exception for email"
+        mock_main_inner.side_effect = Exception(exception_string)
+        # run it
+        options = MockOptions()
+        unattended_upgrade.LOCK_FILE = "./u-u.lock"
+        exception_raised = False
+        try:
+            unattended_upgrade.main(options)
+        except Exception as e:
+            self.assertTrue(str(e) == exception_string)
```

Comment:
You could self.assertEqual() here (not that it matters much). Or even `with self.assertRaises(my_exception):\n...run_thing`.

Reply:
@mvo5  I tried assertRaises first, but it catches the first exception raised in the code even when it is handled and the code raises a different one finally.
AssertEqual is nicer indeed, I'll fix that.